{"status":"success","errorMsg":"","defects":[{"buggyFilePath":"python_security_empty.py","bugId":"5c374dc7-6700-42de-94da-518983c8aea3","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'status' at line [108, 111, 133, 142].MaintainBiz.register_user(duplicate-string)","mainBuggyLine":95,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":93,"codeContext":"        return username in MaintainBiz.today_database\n\n    @staticmethod\n    def register_user(**user_info):\n        \"\"\" Automatically register a new account for user","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"71d81fef-3571-46be-a7fa-ae290b4c1cfc","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'fail' at line [108, 111, 133].MaintainBiz.register_user(duplicate-string)","mainBuggyLine":95,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":93,"codeContext":"        return username in MaintainBiz.today_database\n\n    @staticmethod\n    def register_user(**user_info):\n        \"\"\" Automatically register a new account for user","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"a8e78ef6-3089-4046-aae9-94f0a5be0217","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'message' at line [108, 111, 133, 142].MaintainBiz.register_user(duplicate-string)","mainBuggyLine":95,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":93,"codeContext":"        return username in MaintainBiz.today_database\n\n    @staticmethod\n    def register_user(**user_info):\n        \"\"\" Automatically register a new account for user","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"0bd730d8-6280-4fbd-93ee-f0cf972ca903","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'status' at line [155, 159, 165, 166, 167].MaintainBiz.query_user_info(duplicate-string)","mainBuggyLine":144,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":142,"codeContext":"        return {'status': 'success', 'message': \"Successfully create a account with: \"+str(user_data)}\n\n    @staticmethod\n    def query_user_info(user_id):\n        \"\"\" Propose a query of user information","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"86ffc8c8-c5e4-47f0-ac06-98a80c02915d","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'fail' at line [155, 159, 165, 167].MaintainBiz.query_user_info(duplicate-string)","mainBuggyLine":144,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":142,"codeContext":"        return {'status': 'success', 'message': \"Successfully create a account with: \"+str(user_data)}\n\n    @staticmethod\n    def query_user_info(user_id):\n        \"\"\" Propose a query of user information","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"8a83e7b7-3ba3-4cf2-bb37-f425735b0137","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'message' at line [155, 159, 165, 166, 167].MaintainBiz.query_user_info(duplicate-string)","mainBuggyLine":144,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":142,"codeContext":"        return {'status': 'success', 'message': \"Successfully create a account with: \"+str(user_data)}\n\n    @staticmethod\n    def query_user_info(user_id):\n        \"\"\" Propose a query of user information","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"95e69f91-117a-4f4a-a620-681c8e89ff0b","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'status' at line [181, 182, 186, 193, 195].MaintainBiz.upload_script(duplicate-string)","mainBuggyLine":169,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":167,"codeContext":"        return {'status': 'fail', 'message': \"User not found\"}\n\n    @staticmethod\n    def upload_script(**kwargs):\n        \"\"\" Upload the zip of the script, and unzip to the content","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"7c716a51-dd89-4500-9858-97bae78e51c1","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'fail' at line [181, 182, 186, 193].MaintainBiz.upload_script(duplicate-string)","mainBuggyLine":169,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":167,"codeContext":"        return {'status': 'fail', 'message': \"User not found\"}\n\n    @staticmethod\n    def upload_script(**kwargs):\n        \"\"\" Upload the zip of the script, and unzip to the content","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"e77515d1-dcc1-4be4-8b07-30e1f8850474","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'message' at line [182, 186, 193, 196].MaintainBiz.upload_script(duplicate-string)","mainBuggyLine":169,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":167,"codeContext":"        return {'status': 'fail', 'message': \"User not found\"}\n\n    @staticmethod\n    def upload_script(**kwargs):\n        \"\"\" Upload the zip of the script, and unzip to the content","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"bb8cb5ba-e412-46ab-94ef-d44ab0488f51","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'status' at line [223, 231, 235].MaintainBiz.process_maintain_script(duplicate-string)","mainBuggyLine":210,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":208,"codeContext":"                           f\"User ID: {user_id}, Script: {script_path}, Result: {result}\\n\")\n\n    @staticmethod\n    def process_maintain_script(**kwargs):\n        \"\"\" Execute test tasks","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"49fe51c4-9a9e-4fe5-b208-7190366d6a82","defectLevel":null,"defectType":"H3103","analyzerName":"pylint","description":"Duplicate string literal 'message' at line [223, 231, 235].MaintainBiz.process_maintain_script(duplicate-string)","mainBuggyLine":210,"mainBuggyCode":"    @staticmethod","column":0,"codeContextStartLine":208,"codeContext":"                           f\"User ID: {user_id}, Script: {script_path}, Result: {result}\\n\")\n\n    @staticmethod\n    def process_maintain_script(**kwargs):\n        \"\"\" Execute test tasks","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"f49d6655-719e-42cc-8f31-038dc5c5768e","defectLevel":null,"defectType":"H2101","analyzerName":"pylint","description":"Method name \"do_GET\" doesn't conform to snake_case naming styleMyHTTPRequestHandler.do_GET(huawei-invalid-name)","mainBuggyLine":246,"mainBuggyCode":"    def do_GET(self):","column":0,"codeContextStartLine":244,"codeContext":"    }\n\n    def do_GET(self):\n        url_result = urlsplit(self.path)\n        url_path = url_result.path","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"168cb7e3-c1f8-46f7-98bf-22fbfcdf3e69","defectLevel":null,"defectType":"H2101","analyzerName":"pylint","description":"Method name \"do_POST\" doesn't conform to snake_case naming styleMyHTTPRequestHandler.do_POST(huawei-invalid-name)","mainBuggyLine":264,"mainBuggyCode":"    def do_POST(self):","column":0,"codeContextStartLine":262,"codeContext":"        self.wfile.write(response.getvalue())\n\n    def do_POST(self):\n        content_length = int(self.headers['Content-Length'])\n        input_body = json.loads(","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"1d19547c-7ea3-4681-8a63-bf3d0b5eaff5","defectLevel":null,"defectType":"C0411","analyzerName":"pylint","description":"standard import \"from datetime import datetime, timezone\" should be placed before \"import bcrypt\"(wrong-import-order)","mainBuggyLine":13,"mainBuggyCode":"from datetime import datetime, timezone","column":0,"codeContextStartLine":11,"codeContext":"import ssl\nimport bcrypt\nfrom datetime import datetime, timezone\nfrom http.server import BaseHTTPRequestHandler, HTTPServer\nfrom io import BytesIO","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"e40c044e-f0e9-41ac-a2cc-acc4782b86e7","defectLevel":null,"defectType":"C0411","analyzerName":"pylint","description":"standard import \"from http.server import BaseHTTPRequestHandler, HTTPServer\" should be placed before \"import bcrypt\"(wrong-import-order)","mainBuggyLine":14,"mainBuggyCode":"from http.server import BaseHTTPRequestHandler, HTTPServer","column":0,"codeContextStartLine":12,"codeContext":"import bcrypt\nfrom datetime import datetime, timezone\nfrom http.server import BaseHTTPRequestHandler, HTTPServer\nfrom io import BytesIO\nfrom urllib.parse import parse_qs, urlsplit","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"e4a9fce4-43ae-4fa4-950e-84790dd1d797","defectLevel":null,"defectType":"C0411","analyzerName":"pylint","description":"standard import \"from io import BytesIO\" should be placed before \"import bcrypt\"(wrong-import-order)","mainBuggyLine":15,"mainBuggyCode":"from io import BytesIO","column":0,"codeContextStartLine":13,"codeContext":"from datetime import datetime, timezone\nfrom http.server import BaseHTTPRequestHandler, HTTPServer\nfrom io import BytesIO\nfrom urllib.parse import parse_qs, urlsplit\n","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]},{"buggyFilePath":"python_security_empty.py","bugId":"93aa18d0-6540-4d39-a204-bb7effaa41af","defectLevel":null,"defectType":"C0411","analyzerName":"pylint","description":"standard import \"from urllib.parse import parse_qs, urlsplit\" should be placed before \"import bcrypt\"(wrong-import-order)","mainBuggyLine":16,"mainBuggyCode":"from urllib.parse import parse_qs, urlsplit","column":0,"codeContextStartLine":14,"codeContext":"from http.server import BaseHTTPRequestHandler, HTTPServer\nfrom io import BytesIO\nfrom urllib.parse import parse_qs, urlsplit\n\nLOG_FORMAT = \"%(asctime)s - %(levelname)s - %(message)s\"","mergeKey":null,"extra":"","language":"python","ruleUrl":null,"functionName":null,"engine":null,"category":null,"subcategory":null,"issueType":null,"cwe":null,"extraDegradation":"","extraMsg":"","events":[],"fixCode":[]}]}